CMakeLists.txt

```diff
@@ -81,23 +75,38 @@ else()
   message(FATAL_ERROR "Unsupported PHP version '${PHP_VERSION_ID}'.")
 endif()
 
-add_subdirectory(zai_sapi)
+include_directories("${PHP_PREFIX_PATH}/include/php/"
+        "${PHP_PREFIX_PATH}/include/php/TSRM"
+        "${PHP_PREFIX_PATH}/include/php/Zend"
+        "${PHP_PREFIX_PATH}/include/php/ext"
+        "${PHP_PREFIX_PATH}/include/php/main"
+        "zend_abstract_interface"
+        "ext/vendor/mpack"
+        "ext/vendor/mt19937"
+        "src/dogstatsd"
+        "ext/${PHP_VERSION_DIRECTORY}")
```

Comment:
But we need the ZAI sapi to test ... so we have a direct dependency on ZAI. Unless you explain to me how to properly break that up so that I can use ZAI SAPI in my priority sampling tests?

For me the hierarchy is component > ZAI > ddtrace. i.e. ZAI modules may use components, but not the opposite.

Reply:
Also this really is just the include paths. which technically could be moved into the priority sampling cmake file. It doesn't link anything in which it shouldn't.

Reply:
I think this highlights that we need to move ZAI SAPI out of the `zend_abstract_interface/` directory so that it can be shared more easily. I vaguely remember some discussions @morrisonlevi and I had a while back about how we might support parts of the tracer that depend on both ZAI and ext, but we didn't come to any solid conclusions.

I think it might make sense to have a discussion about this one to find a good path forward.

Reply:
TBH I prefer a more monolithic approach here, just to share the complexity and keep things simpler (in my eyes). I'm not an expert in cmakelists. I can read docs, have set up like 2-3 cmakelists projects over the last 3 years. But overall, if someone has a proposal, feel free to add to this doc, but I personally better don't touch it - at least not in this PR.